pkg/gardenlet/apis/config/validation/validation.go

```diff
@@ -78,13 +80,29 @@ func ValidateGardenletConfiguration(cfg *config.GardenletConfiguration, fldPath
 		}
 	}
 
+	sniPath := fldPath.Child("sni", "ingress")
+	if cfg.SNI != nil && cfg.SNI.Ingress != nil && cfg.SNI.Ingress.ServiceExternalIP != nil {
+		if ip := net.ParseIP(*cfg.SNI.Ingress.ServiceExternalIP); ip == nil {
+			allErrs = append(allErrs, field.Invalid(sniPath.Child("serviceExternalIP"), cfg.SNI.Ingress.ServiceExternalIP, "external service ip is invalid"))
+		}
+	}
+
 	exposureClassHandlersPath := fldPath.Child("exposureClassHandlers")
 	for i, handler := range cfg.ExposureClassHandlers {
 		handlerPath := exposureClassHandlersPath.Index(i)
 
 		for _, errorMessage := range validation.IsDNS1123Label(handler.Name) {
 			allErrs = append(allErrs, field.Invalid(handlerPath.Child("name"), handler.Name, errorMessage))
 		}
+
+		if handler.SNI != nil && handler.SNI.Ingress != nil && handler.SNI.Ingress.ServiceExternalIP != nil {
+			if !cfg.FeatureGates[string(features.APIServerSNI)] {
+				allErrs = append(allErrs, field.Invalid(handlerPath.Child("sni", "ingress", "serviceExternalIP"), handler.SNI.Ingress.ServiceExternalIP, "cannot use an external service ip when APIServerSNI feature gate is disabled"))
```

Comment:
Probably `field.Forbidden` is more appropriate

Reply:
done